src/ir_def/check_ir.ml

```diff
@@ -405,7 +405,9 @@ let rec check_exp env (exp:Ir.exp) : unit =
   | PrimE (p, es) ->
     List.iter (check_exp env) es;
     begin match p, es with
-    | CallPrim insts, [exp1; exp2] ->
+    | CallPrim (insts, par), [exp1; exp2] ->
+      check_exp { env with async = None } par;
```

Comment:
Is this the correct way to ensure that there is no send capability in this part?

Reply:
Do you actually need to?